equatable_gen/lib/src/visitors/class_visitor.dart

```diff
@@ -21,7 +21,7 @@ class ClassVisitor extends RecursiveElementVisitor<void> {
       bool canInclude = true;
 
       for (final exclude in settings.include) {
-        if (RegExp(exclude).hasMatch(element.name)) {
+        if (RegExp(exclude).hasMatch(element.name3!)) {
```

Comment:
Instead of using a non-null assertion here, we can use dart patterns

```suggestion
        if (element.name3 case final String name when RegExp(exclude).hasMatch(name)) {
```